File: android/src/main/java/name/zeno/android/widget/adapter/RcvAdapterItem.java
package name.zeno.android.widget.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import kale.adapter.item.AdapterItem;

/**
 * Create Date: 16/7/4
 *
 * @author 陈治谋 (513500085@qq.com)
 */
public class RcvAdapterItem extends RecyclerView.ViewHolder
{

  public final AdapterItem item;

  public RcvAdapterItem(Context context, ViewGroup parent, AdapterItem item)
  {
    super(LayoutInflater.from(context).inflate(item.getLayoutResId(), parent, false));
    this.item = item;
    this.item.bindViews(itemView);
    this.item.setViews();
  }
}

File: library/src/main/java/name/zeno/android/widget/adapter/RcvAdapterItem.kt
package name.zeno.android.widget.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kale.adapter.item.AdapterItem

/**
 * Create Date: 16/7/4
 *
 * @author 陈治谋 (513500085@qq.com)
 */
class RcvAdapterItem(
    context: Context, parent: ViewGroup, val item: AdapterItem<*>
) : RecyclerView.ViewHolder(LayoutInflater.from(context).inflate(item.layoutResId, parent, false)) {

  init {
    this.item.bindViews(itemView)
    this.item.setViews()
  }
}

